restli-server/src/main/java/com/linkedin/restli/server/ResourceContext.java

```diff
@@ -220,4 +220,15 @@
    * @return whether the request specifies that the resource should return an entity
    */
   boolean isReturnEntityRequested();
-}
\ No newline at end of file
+
+  /**
+   * This is to determine whether the service will fill in default values when the returned value record has some missing
+   * field. In case:
+   * 1. the client is requesting that the default value to be returned. Reads the appropriate
+   * query parameter to determine this information, defaults to false if the query parameter isn't present.
+   * 2. the server is configured to return default value for every request-response pair
+   * @return true if the service need to fill in default value when the field of the return value record is absent
+   */
+  boolean isDefaultValueFillInRequested();
```

Comment:
This is a mouthful. I'd suggest renaming to `isFillDefaultsRequested` or `isFillInDefaultsRequested`